### PR TITLE
feat: reminder improvements

### PR DESCRIPTION
**Pull request by Xazin**
Relates: #3596 

### Feature Preview

Will add a preview soon.

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/user/application/notification_filter/notification_filter_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  }) =>
+      NotificationFilterState(
+        showUnreadsOnly: showUnreadsOnly ?? this.showUnreadsOnly,
+        groupByDate: groupByDate ?? this.groupByDate,
+        sortBy: sortBy ?? this.sortBy,
+      );
+
+  @override
+  List<Object?> get props => [showUnreadsOnly, groupByDate, sortBy];
+}
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/user/application/notification_filter/notification_filter_bloc.dart:75]: Is there any reason for not using the "frozen" syntax here?

File: frontend/appflowy_flutter/lib/user/application/notification_filter/notification_filter_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  ascending,
+}
+
+@freezed
+class NotificationFilterEvent with _$NotificationFilterEvent {
+  const factory NotificationFilterEvent.update({
+    bool? showUnreadsOnly,
+    bool? groupByDate,
+    NotificationSortOption? sortBy,
+  }) = _Update;
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/user/application/notification_filter/notification_filter_bloc.dart:40]: Should we consider splitting this interface into three separate interfaces?

- updateShow...
- updateGroup...
- updateSort...

File: frontend/appflowy_flutter/lib/workspace/presentation/notifications/notification_grouped_view.dart [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  });
+
+  final Map<DateTime, List<ReminderPB>> groupedReminders;
+  final ReminderBloc reminderBloc;
+  final List<ViewPB> views;
+  final PopoverMutex mutex;
+  final bool isUpcoming;
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/workspace/presentation/notifications/notification_grouped_view.dart:32]: This widget tree is too long. We are considering dividing it into smaller widgets.

Comment (Xazin) [reply frontend/appflowy_flutter/lib/workspace/presentation/notifications/notification_grouped_view.dart:32]: I'll abstract it further and simplify it a bit

File: frontend/appflowy_flutter/lib/user/application/notification_filter/notification_filter_bloc.dart (hunk 1 above)
Comment (Xazin) [reply frontend/appflowy_flutter/lib/user/application/notification_filter/notification_filter_bloc.dart:75]: I'll use freezed here :+1:

**Comment by LucasXu0** [comment]
# preview
Screenshot 2023-10-12 at 10 19 48
Screenshot 2023-10-12 at 10 19 52
Screenshot 2023-10-12 at 10 19 56

**Pull request merged**